### PR TITLE
Initial commit include Whole Controller & webhook logic 

### PR DESCRIPTION
**Pull request by wangyikewxgm**
Initial commit include Whole Controller & webhook logic

### CODE REVIEW

**Comment by barnettZQG** [comment]
@wangyikewxgm  Why push the vendor dir?

**Comment by wangyikewxgm** [comment]
> @wangyikewxgm Why push the vendor dir?

Removed

File: PROJECT [hunk 1]
```diff
@@ -0,0 +1,19 @@
+domain: oam.dev
+layout:
+- go.kubebuilder.io/v3
+projectName: vector-controller
+repo: oam-dev/vector-controller
```
Comment (wonderflow) [inline PROJECT]:
```suggestion
repo: kubevela/vector-controller
```

File: PROJECT [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+resources:
+- api:
+    crdVersion: v1
+    namespaced: true
+  controller: true
+  domain: oam.dev
+  group: vector
+  kind: Config
+  path: oam-dev/vector-controller/api/v1alpha1
```
Comment (wonderflow) [inline PROJECT]: kubevela/

File: api/v1alpha1/webhook_suite_test.go [hunk 3]
```diff
@@ -0,0 +1,133 @@
+/*
+Copyright 2022.
```
Comment (wonderflow) [inline api/v1alpha1/webhook_suite_test.go:2]:
```suggestion
Copyright 2022 The KubeVela Authors.
```

File: config/crd/bases/vector.oam.dev_configs.yaml [hunk 4]
```diff
@@ -0,0 +1,72 @@
+---
+apiVersion: apiextensions.k8s.io/v1
```
Comment (wonderflow) [inline config/crd/bases/vector.oam.dev_configs.yaml]: does the config folder necessary? we're using helm chart instead, right?

**Review by wonderflow — COMMENTED** [review]
please also remove the binary [vector-controller-0.1.0.tgz](https://github.com/kubevela/vector-controller/pull/1/files#diff-51c1cc28c9101dbe0ff664e8ed74f9f4a084eae570ba305195a49f0e62a2b572)

File: PROJECT (hunk 1 above)
Comment (wangyikewxgm) [reply PROJECT]: fixed

**Review by wonderflow — APPROVED** [review]

**Pull request merged**